connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/adhoctoolsschema/schema/AdHocToolSchemaGeneratorImpl.java

```diff
@@ -86,4 +96,53 @@ public Map<String, Object> generateToolSchema(AdHocToolElement element) {
 
     return inputSchema;
   }
+
+  private String parameterName(AdHocToolElement element, AdHocToolElementParameter parameter) {
+    final var parameterName = parameter.name();
+    if (StringUtils.isBlank(parameterName)
+        || !parameterName.startsWith(expectedParameterNamespace)) {
+      throw new AdHocToolSchemaGenerationException(
+          "Failed to generate ad-hoc tool schema for element '%s'. Parameter name '%s' is not part of expected namespace '%s'."
+              .formatted(element.elementId(), parameter.name(), expectedParameterNamespace));
+    }
+
+    final var parameterNameWithoutNamespace =
+        parameterName.substring(expectedParameterNamespace.length());
```

Comment:
`zeebe-tagged-parameters` now returns the whole reference (e.g. `toolCall.myValue`) instead only the last segment. We're free how to handle it, but in this schema generator we're now expecting the "namespace" for a tool call to be under `toolCall.*` and failing otherwise.

This removes ambiguity by establishing a single, valid patterns for tool call parameters while still allowing alternative implementations to handle it differently.